### ISSUE TITLE
Completion of strings does not complete for filennames anymore

### ISSUE THREAD

**Issue by hofingermarkus** [spyder-ide/spyder#6803]
Dear Spyder dev Team,

I remember that Spyder used to complete filenames in strings in earlier versions, just as matlab does (provided that the the file is in the current working directory). However, it doesn't seem as this is still possible in the most recent Spyder (3.2.6, 3.2.8) versions, at least on Ubuntu...

## How to recreate:
Open Spyder on some directory that contains files e.g. 
   \mytest.py
   \mytest_data.csv
type "mytest_<tab>"
Spyder won't complete the filename to "mytest_data.csv" neither in the shell nor in the Editor...

## Package Versions

* Spyder version: 3.2.8 
* Python version: 3.5.4
* Qt version: 5.6.2
* PyQt5 version: 5.6
* Operating system: Linux 4.13.0-36-generic

### Dependencies

```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.21.1 (OK)
numpy >=1.7      :  1.14.1 (OK)
sphinx >=0.6.6   :  1.7.1 (OK)
rope >=0.9.4     :  0.10.7 (OK)
jedi >=0.9.0     :  0.11.1 (OK)
psutil >=0.3     :  5.4.3 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  None (NOK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.2.1 (OK)
pylint >=0.25    :  1.8.2 (OK)
```

**Comment by CAM-Gerlach** [comment]
Thanks for the report. Although I tried it when I first started using Spyder (3.2.2 or so), as I was used  to it from Rstudio, I've never noticed such a feature working, and the above test doesn't do anything on my machine either. Thus, I assumed the feature simply wasn't present.

@ccordoba12 Did this indeed used to be the case? Either way, do we want to turn this into feature request? It would be pretty handy...

**Comment by ccordoba12** [comment]
This doesn't worked in the Editor, although it should work in the console. At least, it works for me.

@CAM-Gerlach, could you test on Windows, please?

**Comment by CAM-Gerlach** [comment]
Yup, works just fine in the console. I'd just never tried it there, sorry.

**Comment by ccordoba12** [comment]
> \mytest.py
\mytest_data.csv

@hofingermarkus, do your file names contain backslashes? Because that could be the problem here.

**Comment by hofingermarkus** [comment]
I think I found a clue!
It works if my string starts with a space. This Time I recorded a Screenshot as GIF.
So it doesn't work neither in the Console nor in the Editor (However I get more code completion for other stuff in the editor). 
But in the end I started to experiment and added spaces before and after the string ( [similar to the bug i recently reported ](https://github.com/spyder-ide/spyder/issues/6700) ) and it suddenly started to work. 
Obviously the resulting string is incorrect, since the filename now starts with a leading space, but at least it gets completed. (I see the same effect on windows and ubuntu on two diffrenent machines)

@ccordoba12  I try to avoid special characters in my filenames, so no - there are no backslashes in the filename... 

![spyder_stringfilenamecompletion](https://user-images.githubusercontent.com/12088794/37727615-9757dc3a-2d38-11e8-96c5-f455914fd9fe.gif)

**Comment by CAM-Gerlach** [comment]
Interesting. Can you reproduce #6700 on your machine, then?

**Comment by ccordoba12** [comment]
This looks like a problem in IPython and not in Spyder. What's your IPython version?

**Comment by hofingermarkus** [comment]
@CAM-Gerlach  Well i filed the #6700 report because it was observed on 3 of my machines. (However, before submitting, I'm only double checking on 2 of them (Windows + 1Linux) not on all 3 of them...)

@ccordoba12 As stated in the list above,  Ipython is at 6.2.1  
It maybe is an Ipython problem - since Spyder doesn't have this feature at all - yet.
However, it would be great if Spyder could provide that filename completion inside strings as well.
It really is a time saver and would make the experience more consistent between Spyder and its main console...

**Comment by ccordoba12** [comment]
I don't know why this problem is happening, sorry. It's working fine for me.

**Comment by hofingermarkus** [comment]
@ccordoba12 
What is your current setup?  I could try to see if this works on my PC - maybe this helps to nail it down...

**Comment by CAM-Gerlach** [comment]
@hofingermarkus Well, doh...silly me.

**Comment by hofingermarkus** [comment]
I could let you guys into one of the machines with teamviewer if that helps...

**Comment by hofingermarkus** [comment]
**I think I found it!** It's a bug in the **Greedy Completer** - once it is turned on, the problems start!

How to reproduce:
- Reset Spyder Settings by removing the folder containing the settings files
- Start Spyder 
         => Test filename completion (works)
- Activate Greedy Compiler in the settings: (preferences->Ipyhton console -> Advanced Settings -> greedy completer
-  Check again 
       => Filename completion is now broken (needs extra space), also triggers bug #6700

**Comment by hofingermarkus** [comment]
As far as the shell is concerned  - this is a bug in the Greedy Completer in the Ipython Console.
https://github.com/ipython/ipython/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+greedy+completer

As far as the editor is concerened - FEATURE REQUEST:
**Please add a filename completion to the editor!**

**Comment by daducci** [comment]
Hi all,

Is there any update on this feature? For data analysis using Spyder, this feature would be really useful! I found that, as suggested somewhere else (I cannot find the link anymore), the "filename autocompletion" using TAB works if a string starts with the character "/". So, would it be possible to simply allow TAB to open the autocompletion window if the string starts with, e.g., "./"? Probably this might be an easy workaround for this.

**Comment by ccordoba12** [comment]
@daducci, path completions are working since Spyder 4.2 (I think):

![imagen](https://user-images.githubusercontent.com/365293/129237415-c6717cc4-6b8c-493c-9c87-ef2135d3ea18.png)

If you're not seeing them is probably because you're using an old version, so please update.

**Comment by daducci** [comment]
Hi @ccordoba12 ,

thanks a lot for your reply. I am using the latest version (5.1.1) and yes, completion works, but it is triggered only when the string starts with the character "/" but not "./", so it works for absolute paths but not relative ones. However, it is most common to look for a filename/path that is local to the current folder. Do you think it would be possible to trigger filename/path completion also on case of a string starting with "./"?

**Comment by ccordoba12** [comment]
Relative path completions also work, as you can see here

![imagen](https://user-images.githubusercontent.com/365293/129496505-c79c0898-20d4-42dd-83e2-a8361cd02e98.png)

But for that you need to create a Spyder project (displayed on the left) so we can tell the backend library that provides code completion for us what the current directory is (i.e. the project's path), so it can provide file completions relative to it.

You can create projects from the Projects menu:

![imagen](https://user-images.githubusercontent.com/365293/129496585-da1f7db2-7f68-481a-9330-493f33b622f3.png)

If you're not working in a project, it's really not possible for us to provide relative completions, sorry.

**Comment by daducci** [comment]
Ah ok, I see it now. Do you think it would e possible, instead, to use the current "working directory" (i.e., the directory used for files and IPython consoles) as the path to be used for relative paths? This value is always available also in case a project is not created. I am just wondering as my typical "use case" is to open `.py` files from the file explorer (Fider.app in my case) with Spyder and execute the code inside it cell by cell.

**Comment by ccordoba12** [comment]
No, it is not possible, as I clearly said above. You need to use a Spyder project.
